src/arc_spice/eval/inference_utils.py

```diff
@@ -0,0 +1,133 @@
+from collections.abc import Callable
+
+import torch
+from sklearn.metrics import hamming_loss, zero_one_loss
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+
+from arc_spice.data.multieurlex_utils import MultiHot
+from arc_spice.eval.translation_error import get_comet_model
+from arc_spice.variational_pipelines.RTC_single_component_pipeline import (
+    RTCSingleComponentPipeline,
+)
+from arc_spice.variational_pipelines.RTC_variational_pipeline import (
+    RTCVariationalPipeline,
+)
+
+
+class ResultsGetter:
+    def __init__(self, n_classes):
+        self.func_map: dict[str, Callable] = {
+            "recognition": self.recognition_results,
+            "translation": self.translation_results,
+            "classification": self.classification_results,
+        }
+        self.comet_model = get_comet_model()
+        self.multihot = MultiHot(n_classes)
+
+    def get_results(
+        self,
+        clean_output: dict[str, dict],
+        var_output: dict[str, dict],
+        test_row: dict[str, list[int]],
+        results_dict: dict[str, dict[str, list]],
+    ):
+        for step_name in clean_output:
+            results_dict = self.func_map[step_name](
+                test_row, clean_output, var_output, results_dict
+            )
+        results_dict["input_data"]["celex_ids"].append(test_row["celex_id"])
+        return results_dict
+
+    def recognition_results(self, test_row, clean_output, var_output, results_dict):
+        assert test_row is not None
```

Comment:
if these are None then the whole process will fail. Should this error be handled instead or are we happy that this should cause program to exit?

Reply:
I'd prefer not to make the args ambiguous. If this isn't ready yet, should it just raise a NotImplementedError?